src/Grpc.AspNetCore.Server/Internal/DefaultGrpcInterceptorActivator.cs

```diff
@@ -0,0 +1,79 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using Grpc.Core.Interceptors;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class DefaultGrpcInterceptorActivator<TInterceptor> : IGrpcInterceptorActivator<TInterceptor> where TInterceptor : Interceptor
+    {
+        private static ObjectFactory _objectFactory;
+        private readonly IServiceProvider _serviceProvider;
+        private bool? _created;
+
+        public DefaultGrpcInterceptorActivator(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+        public Interceptor Create(params object[] args)
+        {
+            Debug.Assert(!_created.HasValue, "Grpc interceptor activator must not be reused.");
+
+            _created = false;
+            var interceptor = _serviceProvider.GetService<TInterceptor>();
+            if (interceptor == null)
+            {
+                if (_objectFactory == null)
+                {
+                    var argumentTypes = new Type[args.Length];
+
+                    for (var i = 0; i < args.Length; i++)
+                    {
+                        argumentTypes[i] = args[i].GetType();
```

Comment:
Need to validate that args are never null. That should happen in `AddInterceptor<T>`. Put a comment here saying that args have already been validated

Reply:
Hmm... I'm not sure if we should just reject null as arguments. For example, we sometimes use nullables to indicate a setting or a lack thereof.

Reply:
How could null map to a value in the activator factory?

What does UseMiddleware do when there is a null argument?

Reply:
Looking at our middleware implementation: https://github.com/aspnet/AspNetCore/blob/master/src/Http/Http.Abstractions/src/Extensions/UseMiddlewareExtensions.cs#L92 we don't disallow nulls.